### DISCUSSION TITLE
What is the best way to solve n-queens?

### DISCUSSION THREAD

**Discussion by ankit-akash** [ankit-akash/DSA#39] [Q&A] [answered]
brief

**Comment by fcasamento** [comment, marked as answer]
Hi @ankit-akash 

In my opinion, I think it could be something like this:
`Note:  I don't know if I wanted it in code, so I put it that way.`

1.  Start in the leftmost column
2. If all queens are placed
    return true
3. Try all rows in the current column. 
   Do following for every tried row.
    3.1 If the queen can be placed safely in this row 
       then mark this [row, column] as part of the 
       solution and recursively check if placing
       queen here leads to a solution.
    3.2 If placing the queen in [row, column] leads to
       a solution then return true.
    3.3 If placing queen doesn't lead to a solution then
       unmark this [row, column] (Backtrack) and go to 
       step (a) to try other rows
4. If all rows have been tried and nothing worked,
   return false to trigger backtracking.

**Reply by fcasamento** [reply]
@ankit-akash Mark this as an **answer** as well, we can help each other. 🙏🏻 Pls
